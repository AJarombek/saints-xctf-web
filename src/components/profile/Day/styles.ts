/**
 * JSS styles for the Day component.
 * @author Andrew Jarombek
 * @since 10/19/2020
 */

import {FontMixins} from "../../../styles/mixins";
import Colors, {FeelColors} from "../../../styles/colors";
import color from "color";

export default {
    day: {
        position: 'relative',
        width: `${(1 / 8) * 100}%`,
        border: '1px solid #555',
        backgroundColor: color(Colors.lightBackground).darken(0.05).hex(),

        '&:before': {
            content: '" "',
            display: 'block',
            width: '100%',
            paddingTop: '100%'
        },

        '& > div': {
            position: 'absolute',
            top: 0,
            left: 0,
            right: 0,
            bottom: 0
        }
    },
    feel: {
        backgroundColor: ({feel}: {feel: number}) => FeelColors[feel - 1],
    },
    currentMonth: {
        backgroundColor: Colors.lightestBackground,
    },
    dayOfMonth: {
        ...FontMixins.robotoSlabBold(),
        margin: 0,
        textAlign: 'start',
        padding: 4
    },
    miles: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',

        position: 'absolute',
        height: '100%',
        width: '100%',
        top: 0,

        '& > p': {
            ...FontMixins.robotoSlab(),
            margin: 0,
            textAlign: 'center'
        }
    },
    '@media screen and (max-width: 1050px)': {
        miles: {
            '& > p': {
                fontSize: 14
            }
        },
        dayOfMonth: {
            fontSize: 15
        }
    },
    '@media screen and (max-width: 950px)': {
        dayOfMonth: {
            fontSize: 14
        }
    },
    '@media screen and (max-width: 900px)': {
        miles: {
            '& > p': {
                fontSize: 13
            }
        }
    }
};
