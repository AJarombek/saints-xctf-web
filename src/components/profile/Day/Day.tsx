/**
 * Component for a day in the calendar.
 * @author Andrew Jarombek
 * @since 10/19/2020
 */

import React from 'react';
import {createUseStyles} from "react-jss";
import styles from "./styles";
import moment from "moment";
import classNames from "classnames";

interface IProps {
    date: moment.Moment;
    monthStart: moment.Moment;
    monthEnd: moment.Moment;
    miles: number;
    feel: number;
}

const useStyles = createUseStyles(styles);

const Day: React.FunctionComponent<IProps> = ({ date, monthStart, monthEnd, miles, feel }) => {
    const classes = useStyles();

    return (
        <div className={classNames(classes.day, date >= monthStart && date <= monthEnd && classes.currentMonth)}>
            <div>
                <p className={classes.dayOfMonth}>{date.date()}</p>
                {miles && (
                    <div className={classes.miles}>
                        <p>{miles}</p>
                        <p>Miles</p>
                    </div>
                )}
            </div>
        </div>
    );
};

export default Day;
