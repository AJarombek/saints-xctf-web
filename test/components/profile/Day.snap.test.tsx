/**
 * Snapshot test for the Day component.
 * @author Andrew Jarombek
 * @since 3/6/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import Day from '../../../src/components/profile/Day/Day';
import moment from 'moment';

describe('Day Snapshot Tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Day
          date={moment('2021-03-06')}
          monthStart={moment('2021-03-01')}
          monthEnd={moment('2021-03-31')}
          feel={6}
          miles={5.4}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
