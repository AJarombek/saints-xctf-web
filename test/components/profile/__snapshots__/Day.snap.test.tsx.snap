// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day Snapshot Tests renders correctly 1`] = `
<div
  className="day-0-1-1 currentMonth-0-1-3 feel-0-1-2 feel-d0-0-1-6"
  data-cypress="day"
>
  <div>
    <p
      className="dayOfMonth-0-1-4"
    >
      6
    </p>
    <div
      className="miles-0-1-5"
    >
      <p>
        5.40
      </p>
      <p>
        Miles
      </p>
    </div>
  </div>
</div>
`;
